### PR TITLE
Add config for disabling "View DRD" button

### LINKED ISSUE #349: Be able to disable 'Show DRD' button via config
Example: 

```javscript
new DmnJS({
  modules: [],
  drd: {
    drillDown: {
      enabled: true
    }
  },
  decisionTable: {
    showDrd: {
      enabled: false
    }
  },
  literalExpression: {
    showDrd: {
      enabled: false
    }
  }
})
```

### PR DESCRIPTION
**Pull request by MxCvlibr**
__Which issue does this PR address?__

Closes #349

__Acceptance Criteria__

* [x] Corresponds to the concept 
* [x] Corresponds to the design 

__Definition of Done__

* [x] corresponds to [the design principles](https://github.com/bpmn-io/design-principles)
* [x] corresponds to [the code standards](https://github.com/bpmn-io/bpmn-js/blob/master/.github/CONTRIBUTING.md#creating-a-pull-request)
* [x] passes Continuous Integration checks
* [x] available as feature branch on GitHub
  * [x] contains the cleaned up commit history
  * [x] commit messages satisfy our [commit message guidelines](https://www.conventionalcommits.org/)
  * [x] a single commit closes the issue via Closes #issuenr

### CODE REVIEW

File: packages/dmn-js-decision-table/src/features/view-drd/ViewDrd.js [hunk 1]
```diff
@@ -1,12 +1,13 @@
 import ViewDrdComponent from './components/ViewDrdComponent';
 
 export default class ViewDrd {
-  constructor(components, eventBus, injector, sheet) {
+  constructor(components, eventBus, injector, sheet, config) {
```
Comment (pinussilvestrus) [inline packages/dmn-js-decision-table/src/features/view-drd/ViewDrd.js:4]: this will always be `undefined` because you did not declare `config` in the injected modules, cf. https://github.com/bpmn-io/dmn-js/blob/b65fe68c1cc09822904dde5864248835199697ae/packages/dmn-js-decision-table/src/features/view-drd/ViewDrd.js#L48

Comment (pinussilvestrus) [inline packages/dmn-js-decision-table/src/features/view-drd/ViewDrd.js:4]: Did you consider to also add this behavior to other `viewDRD` / `drillDown` components, as the issue description reveals?

cf. [`dmn-js-literal-expression@ViewDrd`](https://github.com/bpmn-io/dmn-js/blob/develop/packages/dmn-js-literal-expression/src/features/view-drd/ViewDrd.js)
cf. [`dmn-js-drd@DrillDown`](https://github.com/bpmn-io/dmn-js/blob/develop/packages/dmn-js-drd/src/features/drill-down/DrillDown.js#L127)

File: packages/dmn-js-decision-table/src/features/view-drd/ViewDrd.js [hunk 2]
```diff
@@ -1,12 +1,13 @@
...
 export default class ViewDrd {
-  constructor(components, eventBus, injector, sheet) {
+  constructor(components, eventBus, injector, sheet, config) {
     this._injector = injector;
     this._sheet = sheet;
+    this._config = config || { enabled: true };
 
     components.onGetComponent('table.before', () => {
-      if (this.canViewDrd()) {
+      if (this.canViewDrd() && this._config.enabled) {
```
Comment (pinussilvestrus) [inline packages/dmn-js-decision-table/src/features/view-drd/ViewDrd.js:10]: Can you think of any test case which verifies the added behavior?

**Review by pinussilvestrus — CHANGES_REQUESTED** [review]
There are difficulties with the solution you proposed. Also, make sure your PR fits our definition of done, in terms of[ commit message guidelines](https://www.conventionalcommits.org/).

**Comment by pinussilvestrus** [comment]
@MxCvlibr any updates on this? Do you need help?

**Comment by MxCvlibr** [comment]
> @MxCvlibr any updates on this? Do you need help?

I'm good, I'll get it done ASAP

File: packages/dmn-js-decision-table/src/features/view-drd/ViewDrd.js (hunk 1 above)
Comment (MxCvlibr) [reply packages/dmn-js-decision-table/src/features/view-drd/ViewDrd.js:4]: After adding 'config', there's an error from the test written before: "view drd should show view drd button FAILED" AssertionError: expected null to exist, so I think the domQuery for the view-drd-button returns null. I don't know if I missed anything from the injected module, config part or the test? And do the MockViewer vs DmnDecisionTableViewer have anything to do with this problem?

Comment (MxCvlibr) [reply packages/dmn-js-decision-table/src/features/view-drd/ViewDrd.js:4]: How do I interact with the DrillDown in localhost to see its behavior?

Comment (pinussilvestrus) [reply packages/dmn-js-decision-table/src/features/view-drd/ViewDrd.js:4]: Can you reference the test case you mean?

Comment (pinussilvestrus) [reply packages/dmn-js-decision-table/src/features/view-drd/ViewDrd.js:4]: Sometimes it makes sense to mark one test case with `only` to ensure only this test is running.

> it.only(...)

Otherwise, karma would reuse the container for the next test case and you can't interact with it anymore.

**Comment by pinussilvestrus** [comment]
@MxCvlibr what's the plan for this PR? Do you need help?  I know [that you're currently busy](https://github.com/bpmn-io/dmn-js/pull/588#issuecomment-711998297) 🙂 , but it would be great if we could make some progress here.

**Comment by MxCvlibr** [comment]
> @MxCvlibr what's the plan for this PR? Do you need help? I know [that you're currently busy](https://github.com/bpmn-io/dmn-js/pull/588#issuecomment-711998297) 🙂 , but it would be great if we could make some progress here.

File: packages/dmn-js-decision-table/src/features/view-drd/ViewDrd.js (hunk 1 above)
Comment (MxCvlibr) [reply packages/dmn-js-decision-table/src/features/view-drd/ViewDrd.js:4]: I tried using `only()` but it's not allowed, `"Unexpected exclusive mocha test"`. How do I fix this? I tried using `npm run dev -- dmn-js-drd --grep "should drill down"` as answered on stackoverflow and it still runs all the tests.

Comment (pinussilvestrus) [reply packages/dmn-js-decision-table/src/features/view-drd/ViewDrd.js:4]: The test itself is located in the `dmn-js-decision-table` package, therefore specifing the test with only and executing 

`npm run dev -- dmn-js-decision-table`

in the root folder of the project should work.

Comment (pinussilvestrus) [reply packages/dmn-js-decision-table/src/features/view-drd/ViewDrd.js:4]: Which test are you trying to run with "only"?

Comment (MxCvlibr) [reply packages/dmn-js-decision-table/src/features/view-drd/ViewDrd.js:4]: It says `'karma' is not recognized as an internal or external command`, I tried npm install but didn't work.
PS: Because of my schedule and I'm new to this, if you find these PRs slow and not going anywhere we can close the PRs.

Comment (MxCvlibr) [reply packages/dmn-js-decision-table/src/features/view-drd/ViewDrd.js:4]: I'm trying to test the drilldown part. I'm still not really sure, there are several tests in the DrillDownSpec.js.

Comment (pinussilvestrus) [reply packages/dmn-js-decision-table/src/features/view-drd/ViewDrd.js:4]: > Because of my schedule and I'm new to this, if you find these PRs slow and not going anywhere we can close the PRs.

We don't have to rush. Your PRs (this + https://github.com/bpmn-io/dmn-js/pull/588) tackle valid problems and basically, the solution sketch seems to be right for me.

However, it's up to you how and whether to continue with it. We can just try to support you and give you advice for specific questions you ask.

**Comment by philippfromme** [comment]
I'm closing this PR since there hasn't been any development and the _View DRD_ can already be disabled simply by overriding the module:

```javascript
new DmnJS({
  decisionTable: {
    additionalModules: [
      {
        viewDrd: [ value, null ]
      }
    ]
  },
  literalExpression: {
    additionalModules: [
      {
        viewDrd: [ value, null ]
      }
    ]
  }
});
```

**Comment by pandika-csgi** [comment]
> I'm closing this PR since there hasn't been any development and the _View DRD_ can already be disabled simply by overriding the module:
> 
> ```js
> new DmnJS({
>   decisionTable: {
>     additionalModules: [
>       {
>         viewDrd: [ value, null ]
>       }
>     ]
>   },
>   literalExpression: {
>     additionalModules: [
>       {
>         viewDrd: [ value, null ]
>       }
>     ]
>   }
> });
> ```

Hi @philippfromme, where I can override that? when we create the `DmnJS` instance? I cant find that in my module
Thank you
